dot/sync/chain_sync.go

```diff
@@ -525,11 +522,15 @@ func (cs *chainSync) getTarget() *big.Int {
 		return big.NewInt(2<<32 - 1)
 	}
 
+	// we are going to sort the data and remove the outliers then we will return the avg of all the valid elements
+	intArr := make([]*big.Int, len(cs.peerState))
+	i := 0
 	for _, ps := range cs.peerState {
-		sum = big.NewInt(0).Add(sum, ps.number)
-		count++
+		intArr[i] = ps.number
```

Comment:
nit: use the index given by the `range` as the amount of items in `peerState` is the same on `intArr`
````go
intArr := make([]*big.Int, len(cs.peerState))
for i, ps := range cs.peerState {
	intArr[i] = ps.number
}
```

Reply:
@EclesioMeloJunior `cs.peerState` is a map. So range iter will produce key, value rather than an index value.

Reply:
Oh,ok haha

Reply:
I would suggest you to
`slice := make([]int, 0, len(cs.peerState))`
Then inside the for
`slice = append(slice, ps.number)`
And you don't need to use the `i` variable, and the slice now have the same capacity as cs.peerState however starting empty

Reply:
:laughing: Well I think we had [the same thinking](https://github.com/ChainSafe/gossamer/pull/1992#discussion_r749444946) 5 minutes ago!

Reply:
😃🤷‍♂️